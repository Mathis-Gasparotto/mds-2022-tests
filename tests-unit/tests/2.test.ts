import { expect, test } from 'vitest'
import { calculateTotalTime, User } from '../2'

const users: User[] = [
  {
    name: 'Alice',
    records: [
      { time: 10, name: 'Record 1' },
      { time: 20, name: 'Record 2' }
      // Add more records here...
    ]
  },
  {
    name: 'Bob',
    records: [
      { time: 15, name: 'Record 3' },
      { time: 25, name: 'Record 4' }
      // Add more records here...
    ]
  }
  // Add more users here...
]

test('asc', () => {
  const actual = calculateTotalTime(users, 'asc')

  expect(actual).toMatchSnapshot()
})

test('desc', () => {
  const actual = calculateTotalTime(users, 'desc')

  expect(actual).toMatchSnapshot()
})
