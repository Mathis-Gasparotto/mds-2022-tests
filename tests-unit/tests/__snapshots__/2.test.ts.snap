// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`asc 1`] = `
[
  {
    "name": "Alice",
    "totalTime": 30,
  },
  {
    "name": "Bob",
    "totalTime": 40,
  },
]
`;

exports[`desc 1`] = `
[
  {
    "name": "Bob",
    "totalTime": 40,
  },
  {
    "name": "Alice",
    "totalTime": 30,
  },
]
`;
